### ISSUE TITLE
plugin 'arm64' is not installed error upon following install instructions

### ISSUE THREAD

**Issue by feoh** [open-craft/tutor-contrib-arm64#2]
Hello! I'm trying to get tutor up and running on my M2 Mac running MacOS X Ventura 13.2.1 (latest).

Here is a log of my invocations and output. Shell is zsh + oh-my-zsh but I seriously doubt that has any effect :) Python is 3.10 installed with Homebrew.

```
# cpatti @ rocinante in ~/src/mit [16:28:51] C:130
$ mkdir tutor-arm64

# cpatti @ rocinante in ~/src/mit [16:28:55] 
$ cd tutor-arm64

# cpatti @ rocinante in ~/src/mit/tutor-arm64 [16:28:58] 
$ python3.10 -m venv tutor-arm64-venv

# cpatti @ rocinante in ~/src/mit/tutor-arm64 [16:29:13] 
$ . ./tutor-arm64-venv/bin/activate

# cpatti @ rocinante in ~/src/mit/tutor-arm64 tutor-arm64-venv [16:29:22] 
$ pip install git+https://github.com/open-craft/tutor-contrib-arm64
Collecting git+https://github.com/open-craft/tutor-contrib-arm64
  Cloning https://github.com/open-craft/tutor-contrib-arm64 to /private/var/folders/w3/jmw69rnx10908b_xnfw5rcdr0000gn/T/pip-req-build-zdj1wb6l
  Running command git clone --filter=blob:none --quiet https://github.com/open-craft/tutor-contrib-arm64 /private/var/folders/w3/jmw69rnx10908b_xnfw5rcdr0000gn/T/pip-req-build-zdj1wb6l
  Resolved https://github.com/open-craft/tutor-contrib-arm64 to commit 2d35230d98a2ee35c55e8e18fad24633737ded30
  Preparing metadata (setup.py) ... done
Collecting tutor
  Using cached tutor-15.3.3.tar.gz (89 kB)
  Preparing metadata (setup.py) ... done
Collecting appdirs
  Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting click>=8.0
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting jinja2>=2.10
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting kubernetes
  Using cached kubernetes-26.1.0-py2.py3-none-any.whl (1.4 MB)
Collecting mypy
  Using cached mypy-1.1.1-cp310-cp310-macosx_11_0_arm64.whl (9.7 MB)
Collecting pycryptodome
  Using cached pycryptodome-3.17-cp35-abi3-macosx_10_9_universal2.whl (2.4 MB)
Collecting pyyaml>=6.0
  Using cached PyYAML-6.0-cp310-cp310-macosx_11_0_arm64.whl (173 kB)
Collecting typing-extensions>=4.4.0
  Using cached typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.2-cp310-cp310-macosx_10_9_universal2.whl (17 kB)
Collecting python-dateutil>=2.5.3
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting requests-oauthlib
  Using cached requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting urllib3>=1.24.2
  Using cached urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
Collecting google-auth>=1.0.1
  Downloading google_auth-2.17.0-py2.py3-none-any.whl (178 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 178.1/178.1 kB 3.3 MB/s eta 0:00:00
Requirement already satisfied: setuptools>=21.0.0 in ./tutor-arm64-venv/lib/python3.10/site-packages (from kubernetes->tutor->tutor-contrib-arm64==0.1.0) (67.2.0)
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Collecting websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0
  Using cached websocket_client-1.5.1-py3-none-any.whl (55 kB)
Collecting certifi>=14.05.14
  Using cached certifi-2022.12.7-py3-none-any.whl (155 kB)
Collecting six>=1.9.0
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting mypy-extensions>=1.0.0
  Using cached mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting cachetools<6.0,>=2.0.0
  Using cached cachetools-5.3.0-py3-none-any.whl (9.3 kB)
Collecting rsa<5,>=3.1.4
  Using cached rsa-4.9-py3-none-any.whl (34 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.1.0-cp310-cp310-macosx_11_0_arm64.whl (123 kB)
Collecting oauthlib>=3.0.0
  Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Installing collected packages: pyasn1, appdirs, websocket-client, urllib3, typing-extensions, tomli, six, rsa, pyyaml, pycryptodome, pyasn1-modules, oauthlib, mypy-extensions, MarkupSafe, idna, click, charset-normalizer, certifi, cachetools, requests, python-dateutil, mypy, jinja2, google-auth, requests-oauthlib, kubernetes, tutor, tutor-contrib-arm64
  DEPRECATION: tutor is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for tutor ... done
  DEPRECATION: tutor-contrib-arm64 is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for tutor-contrib-arm64 ... done
Successfully installed MarkupSafe-2.1.2 appdirs-1.4.4 cachetools-5.3.0 certifi-2022.12.7 charset-normalizer-3.1.0 click-8.1.3 google-auth-2.17.0 idna-3.4 jinja2-3.1.2 kubernetes-26.1.0 mypy-1.1.1 mypy-extensions-1.0.0 oauthlib-3.2.2 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycryptodome-3.17 python-dateutil-2.8.2 pyyaml-6.0 requests-2.28.2 requests-oauthlib-1.3.1 rsa-4.9 six-1.16.0 tomli-2.0.1 tutor-15.3.3 tutor-contrib-arm64-0.1.0 typing-extensions-4.5.0 urllib3-1.26.15 websocket-client-1.5.1

[notice] A new release of pip is available: 23.0 -> 23.0.1
[notice] To update, run: pip install --upgrade pip

# cpatti @ rocinante in ~/src/mit/tutor-arm64 tutor-arm64-venv [16:29:33] 
$ tutor plugins enable arm64
Error: plugin 'arm64' is not installed.

```

Happy to contribute any way I can!

**Comment by feoh** [comment]
Hi. I was able to get this plugin to work by building and installing it myself locally.

Exact steps in case anyone else needs this:
* Clone this repo
* Create a virtualenv for this project. e.g. `python -m venv tutor-contrib-arm64-venv` 
* Activate the venv e.g. `. ./tutor-contrib-arm64-vent`
* Build the plugin locally: `python setup.py build`
* Install the plugin locally: `tutor plugin install ./tutor_arm64/plugin.py`

At this point I was able to successfully `tutor  local launch' and connect to my running instance.

**Comment by bradenmacdonald** [comment]
Hi @feoh, thanks for the report. Sorry for the slow reply due to the conference. I will investigate.

**Comment by bradenmacdonald** [comment]
@feoh I can't reproduce your bug report.

```
% mkdir tutor-arm64
% cd tutor-arm64
% python3.10 -m venv tutor-arm64-venv
% . ./tutor-arm64-venv/bin/activate
(tutor-arm64-venv) % pip install tutor
Collecting tutor
  Downloading tutor-15.3.3.tar.gz (89 kB)
...
(tutor-arm64-venv) % pip install git+https://github.com/open-craft/tutor-contrib-arm64
Collecting git+https://github.com/open-craft/tutor-contrib-arm64
  Cloning https://github.com/open-craft/tutor-contrib-arm64 to /private/var/folders/tr/bx5ym9bd1nb3zc24sl6njkrr0000gp/T/pip-req-build-3vm_bj8c
  Running command git clone --filter=blob:none --quiet https://github.com/open-craft/tutor-contrib-arm64 /private/var/folders/tr/bx5ym9bd1nb3zc24sl6njkrr0000gp/T/pip-req-build-3vm_bj8c
  Resolved https://github.com/open-craft/tutor-contrib-arm64 to commit 2d35230d98a2ee35c55e8e18fad24633737ded30
  Preparing metadata (setup.py) ... done
...
Installing collected packages: tutor-contrib-arm64
  DEPRECATION: tutor-contrib-arm64 is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for tutor-contrib-arm64 ... done
Successfully installed tutor-contrib-arm64-0.1.0

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip

(tutor-arm64-venv) % tutor plugins list
NAME 	STATUS   	VERSION
arm64	✅ enabled	0.1.0
(tutor-arm64-venv) % tutor plugins enable arm64
Plugin arm64 enabled
Configuration saved to /Users/braden/Library/Application Support/tutor/config.yml
You should now re-generate your environment with `tutor config save`.
(tutor-arm64-venv) % 
```

I also tried using the exact same version of pip as you, and not installing tutor with a separate command, so I should be using _exactly_ the same sequence of commands as you, yet it was still working every time. Any ideas?

**Comment by feoh** [comment]
Glad to hear! I don't know what to say other than I just tried from a fresh slate and got the exact same result.

Happy to chalk it up to something I'm doing wrong or weirdness in my environment.  🤷
